streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1812,24 +1812,28 @@ class ZStream[-R, +E, +A](val process: ZManaged[R, E, Pull[R, E, A]]) extends Se
         init  <- schedule.initial.toManaged_
         state <- Ref.make[(Boolean, schedule.State, Option[() => B])]((false, init, None)).toManaged_
         pull = state.get.flatMap {
-          case (done, sched, decision0) =>
+          case (done, sched0, finish0) =>
```

Comment:
We're not actually setting `done = true` anywhere aren't we?

Reply:
ha, correct, I'll get rid of it.